src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedProjectAction.cs

```diff
@@ -39,13 +40,28 @@ public class BuildIntegratedProjectAction : NuGetProjectAction
         /// </summary>
         public IReadOnlyList<NuGetProjectAction> OriginalActions { get; }
 
+        /// <summary>
+        /// Shows the frameworks for which a preview restore operation was successful. Only use it
+        /// in case of single package install case, and only for CpsPackageReference projects.
+        /// </summary>
+        public IEnumerable<NuGetFramework> SuccessfulFrameworksForPreviewRestore { get; }
```

Comment:
I think @rohit21agrawal. These properties are designed for the single install scenario. Since we don't have the UI or plan for a multi-install or update scenarios, we shouldn't complicate the model.